src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -7,61 +7,265 @@ import {
   customElement,
 } from "lit-element";
 import "@material/mwc-button";
+import { until } from "lit-html/directives/until";
 
+import "../../../../components/ha-card";
+
+import { CardPickTarget, CardPickerConfig } from "../types";
+import { HuiErrorCard } from "../../cards/hui-error-card";
 import { HomeAssistant } from "../../../../types";
-import { LovelaceCardConfig } from "../../../../data/lovelace";
-import { CardPickTarget } from "../types";
+import { LovelaceCard } from "../../types";
+import { LovelaceCardConfig, LovelaceConfig } from "../../../../data/lovelace";
+import { computeDomain } from "../../../../common/entity/compute_domain";
 import { fireEvent } from "../../../../common/dom/fire_event";
-import { getCardElementClass } from "../../create-element/create-card-element";
-
-const cards: string[] = [
-  "alarm-panel",
-  "conditional",
-  "entities",
-  "button",
-  "entity-filter",
-  "gauge",
-  "glance",
-  "history-graph",
-  "horizontal-stack",
-  "iframe",
-  "light",
-  "map",
-  "markdown",
-  "media-control",
-  "picture",
-  "picture-elements",
-  "picture-entity",
-  "picture-glance",
-  "plant-status",
-  "sensor",
-  "shopping-list",
-  "thermostat",
-  "vertical-stack",
-  "weather-forecast",
+import {
+  getCardElementClass,
+  createCardElement,
+} from "../../create-element/create-card-element";
+import {
+  computeUnusedEntities,
+  computeUsedEntities,
+  EXCLUDED_DOMAINS,
+} from "../../common/compute-unused-entities";
+
+const cardConfigs: CardPickerConfig[] = [
+  {
+    lovelaceCardConfig: { type: "alarm-panel" },
+    includeDomains: ["alarm_control_panel"],
```

Comment:
We should move this to the card as well. So we can add the array with types back instead of the objects.